controllers/toolchaincluster/toolchaincluster_controller.go

```diff
@@ -75,20 +78,52 @@ func (r *Reconciler) Reconcile(ctx context.Context, request ctrl.Request) (ctrl.
 		reqLogger.Error(err, "cannot create ClientSet for the ToolchainCluster")
 		return reconcile.Result{}, err
```

Comment:
if we fail to create the client, then we should make sure that the ready condition is set to false

Reply:
okay

Reply:
I did update it, but testing this functionality doesn't make sense, as we never encounter this scenario.